modules/provision/local_store.go

```diff
@@ -0,0 +1,81 @@
+package provision
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type memStore struct {
+	sync.RWMutex
+	m map[string]*Reservation
+}
+
+func NewMemStore() *memStore {
+	return &memStore{
+		m: make(map[string]*Reservation),
+	}
+}
+
+// Add a reservation ID to the store
+func (s *memStore) Add(r *Reservation) error {
+	s.Lock()
+	defer s.Unlock()
+
+	if _, ok := s.m[r.ID]; ok {
+		return fmt.Errorf("reservation %s already in the store", r.ID)
+	}
+
+	s.m[r.ID] = r
+	return nil
+}
+
+// Remove a reservation ID from the store
+func (s *memStore) Remove(id string) error {
+	s.Lock()
+	defer s.Unlock()
+
+	if _, ok := s.m[id]; ok {
+		delete(s.m, id)
+	}
+
+	return nil
+}
+
+// GetExpired returns all id the the reservations that are expired
+// at the time of the function call
+func (s *memStore) GetExpired() ([]*Reservation, error) {
+	s.RLock()
+	defer s.RUnlock()
+
+	output := make([]*Reservation, 0, len(s.m)/2)
+	for _, r := range s.m {
+		if !isExpired(r) {
+			continue
+		}
+		output = append(output, r)
+	}
+	return output, nil
+}
+
+func isExpired(r *Reservation) bool {
```

Comment:
Why this is not defined as a member method on the Reservation itself? 
like
```suggestion
func (r *Reservation) Expired() bool {
```